### PR TITLE
OSX tester does not pick up /rebuild

### PR DESCRIPTION
**Pull request by marcfehling**
I noticed that the OSX tester sometimes remains in the `Pending — need ready to test label and /rebuild` state even with the proper label and comment.

It's fine when the `ready to test` flag is there when the PR is created, but when added later and issuing a build with `/rebuild` it remains unresponsive, see for example #14995.

It seems to be working when (force) pushing on a branch when the corresponding PR already has the flag, like in #14982.

In an attempt to solve the problem, I tried to align the Jenkinsfiles so that they are almost identical. @tjhei -- Do you think there is another setting you need to adjust on the server after #14902?

### CODE REVIEW

File: contrib/ci/Jenkinsfile.osx [hunk 1]
```diff
@@ -34,15 +34,18 @@ pipeline
...
 
     stage("main")
     {
-      agent {
-        node {
+      agent
+      {
+        node
+        {
           label 'osx'
```
Comment (marcfehling) [inline contrib/ci/Jenkinsfile.osx:49]:
```suggestion
          label 'jenkins: osx'
```
Just a shot in the dark. Maybe this label needs to be adjusted as well?

**Comment by tjhei** [comment]
I was not aware of this issue, thanks. It turns out I forgot to enable the "rebuild" trigger for the osx tester. Oops. This is now fixed.

File: contrib/ci/Jenkinsfile.osx (hunk 1 above)
Comment (tjhei) [reply contrib/ci/Jenkinsfile.osx:49]: No. The machine label is something different

File: README.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ Continuous Integration Status:
...
 | Linux MPI | [![Build Status](https://ci.tjhei.info/job/dealii-mpi/job/master/badge/icon)](https://ci.tjhei.info/job/dealii-mpi/job/master/)                                                                                                                | See https://ci.tjhei.info |
-| MacOS  | [![Build Status](https://jenkins.tjhei.info/job/dealii-OSX/job/master/badge/icon)](https://jenkins.tjhei.info/job/dealii-OSX/job/master/)                                                                                                        | See https://jenkins.tjhei.info                                                        |
+| MacOS  | [![Build Status](https://ci.tjhei.info/job/dealii-OSX/job/master/badge/icon)](https://ci.tjhei.info/job/dealii-OSX/job/master/)                                                                                                        | See https://ci.tjhei.info                                                        |
```
Comment (tjhei) [inline README.md:65]: Thank you!

**Review by tjhei — APPROVED** [review]

**Pull request merged**